pkgs/development/python-modules/cryptography/default.nix

```diff
@@ -46,15 +51,26 @@ buildPythonPackage rec {
 
   cargoRoot = "src/rust";
 
+  env = {
+    PYO3_CROSS_LIB_DIR = "${python3}/lib/";
+    CARGO_BUILD_TARGET = "${rust.toRustTargetSpec stdenv.hostPlatform}";
+  };
+
   nativeBuildInputs = lib.optionals (!isPyPy) [
     cffi
   ] ++ [
-    rustPlatform.cargoSetupHook
-    setuptools-rust
-  ] ++ (with rustPlatform; [ rust.cargo rust.rustc ]);
+    cargo
+    rustc
+  ];
 
-  buildInputs = [ openssl ]
-    ++ lib.optionals stdenv.isDarwin [ Security libiconv ];
+  buildInputs = [
+    openssl
+    setuptools-rust
+    rustPlatform.cargoSetupHook
```

Comment:
setuptools is consumed by the python that is building as far as I know. setup hooks also usually go to nativeBuildInputs. I assume the hooks themselves require some fixes to fix this properly.